data/src/main/java/com/ericsson/bss/cassandra/ecchronos/data/sync/EccNodesSync.java

```diff
@@ -0,0 +1,275 @@
+        myUpdateStatusStatement = mySession.prepare(QueryBuilder.update(KEYSPACE_NAME, TABLE_NAME)
+                .setColumn(COLUMN_NODE_STATUS, bindMarker())
+                .setColumn(COLUMN_LAST_CONNECTION, bindMarker())
+                .setColumn(COLUMN_NEXT_CONNECTION, bindMarker())
+                .whereColumn(COLUMN_ECCHRONOS_ID).isEqualTo(bindMarker())
+                .whereColumn(COLUMN_DC_NAME).isEqualTo(bindMarker())
+                .whereColumn(COLUMN_NODE_ID).isEqualTo(bindMarker())
+                .build()
+                .setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM));
+        ecChronosID = builder.myEcchronosID;
+    }
+
+    public void acquireNodes() throws EcChronosException
+    {
+        if (myNodesList.isEmpty())
+        {
+            throw new EcChronosException("Cannot Acquire Nodes because there is no nodes to be acquired");
+        }
+        for (Node node : myNodesList)
+        {
+            LOG.info(
+                    "Preparing to acquire node {} with endpoint {} and Datacenter {}",
+                    node.getHostId(),
+                    node.getEndPoint(),
+                    node.getDatacenter());
+            ResultSet tmpResultSet = acquireNode(node);
+            if (tmpResultSet.wasApplied())
+            {
+                LOG.info("Node successfully acquired by instance {}", ecChronosID);
+            }
+            else
+            {
+                LOG.error("Unable to acquire node {}", node.getHostId());
+            }
+        }
+    }
+
+    private ResultSet acquireNode(final Node node)
+    {
+        return insertNodeInfo(
+                node.getDatacenter(),
+                node.getEndPoint().toString(),
+                node.getState().toString(),
+                Instant.now(),
+                Instant.now().plus(DEFAULT_CONNECTION_DELAY_IN_MINUTES, ChronoUnit.MINUTES),
+                node.getHostId());
+    }
+
+    @VisibleForTesting
+    public ResultSet verifyAcquireNode(final Node node)
+    {
+        return acquireNode(node);
+    }
+
+    private ResultSet insertNodeInfo(
+            final String datacenterName,
+            final String nodeEndpoint,
+            final String nodeStatus,
+            final Instant lastConnection,
+            final Instant nextConnection,
+            final UUID nodeID
+    )
+    {
+        BoundStatement insertNodeSyncInfo = myCreateStatement.bind(ecChronosID,
+                datacenterName, nodeEndpoint, nodeStatus, lastConnection, nextConnection, nodeID);
+        return execute(insertNodeSyncInfo);
+    }
+
+    public ResultSet updateNodeStatus(
+            final NodeStatus nodeStatus,
+            final String datacenterName,
+            final UUID nodeID
+    )
+    {
+        ResultSet tmpResultSet = updateNodeStateStatement(nodeStatus, datacenterName, nodeID);
+        if (tmpResultSet.wasApplied())
+        {
+            LOG.info("Node {} successfully upgraded", nodeID);
```

Comment:
upgraded? Do you actually mean updated? ... Or have we built in something secret? ;)

Reply:
:D